### PR TITLE
Simplified code for docs

### PR DESCRIPTION
**Pull request by levhitaIncode**
* migrated code to use backend
* migrated code to use npm version of incode

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,14 +3,17 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@incodetech/welcome": "1.69.1",
```
Comment (levhitaIncode) [inline package.json:6]: only works up to this version

File: package.json [hunk 2]
```diff
@@ -3,14 +3,17 @@
...
+    "react-dom": "^18.3.1",
     "@testing-library/jest-dom": "^5.17.0",
     "@testing-library/react": "^13.4.0",
     "@testing-library/user-event": "^13.5.0",
     "react-scripts": "5.0.1",
     "web-vitals": "^2.1.4"
   },
   "scripts": {
-    "start": "HTTPS=true SSL_CRT_FILE=./.cert/cert.pem SSL_KEY_FILE=./.cert/key.pem react-scripts start",
+    "start": "HTTPS=true PORT=5173 HOST=0.0.0.0 react-scripts start",
```
Comment (levhitaIncode) [inline package.json:16]: no need to add cert files
expose server to the local network (for mobile testing)

File: .env.example [hunk 3]
```diff
@@ -1,14 +1,5 @@
...
-# Web SDK URL. Check our latest version here:
-# https://developer.incode.com/docs/releases-web-sdk
-REACT_APP_INCODE_WEB_SDK_URL="https://sdk.incode.com/sdk/onBoarding-1.71.0.js"
-
-# Dev server config values (only for localhost).
-REACT_APP_INCODE_OMNI_PATH="/omni"
-REACT_APP_SDK_ENDPOINT=""
\ No newline at end of file
+# URL of your backend server that does `/start`
+REACT_APP_TOKEN_SERVER_URL=/api
```
Comment (levhitaIncode) [inline .env.example]: I tried to add the reverse proxy that we use in vite, no success yet, instead I added the backend server thing to the readme and a mention to ngrok to expose it.

File: src/App.js [hunk 4]
```diff
@@ -1,19 +1,32 @@
...
-  MORE INFO:
-  https://developer.incode.com/docs/tutorial-creating-an-identity-validation-app
+MORE INFO:
+https://developer.incode.com/docs/tutorial-creating-an-identity-validation-app
 */
 
 import React, { useEffect, useState, useRef } from "react";
-import { getIncodeToken } from "./incode-session-service";
-import translations from "./translations";
+import { incode } from "./incode";
```
Comment (levhitaIncode) [inline src/App.js:9]: importing from npm, moved translations inside a new file that create the incode object.

File: src/App.js [hunk 5]
```diff
@@ -1,19 +1,32 @@
...
 import "./App.css";
-let incode;
+
 let incodeSession;
-let SDKActive = false;
 let container;
 
+//Function to fetch the onboarding session from your backend
+async function startOnboardingSession() {
```
Comment (levhitaIncode) [inline src/App.js:16]: smaller function that calls the backend instead of trying to create the session here

File: src/App.js [hunk 6]
```diff
@@ -54,9 +67,11 @@ function captureSelfie() {
     showTutorial: true,
   });
 }
+
 function finishOnboarding() {
   incode.getFinishStatus(null, { token: incodeSession.token }).then(() => {
     console.log("Onboarding Finished");
+    container.innerHTML="Onboarding Finished";
```
Comment (levhitaIncode) [inline src/App.js:74]: give visual feedback to the enduser directly in the page

File: src/App.js [hunk 7]
```diff
@@ -66,51 +81,44 @@ function saveDeviceData() {
...
+  const [session, setSession] = useState(null); // Stores the Session
   const incodeContainerRef = useRef(null);
-  const [scriptLoaded, setScriptLoaded] = useState(false);
-
+  
+  // Store data that will not trigger re-renders unless specifically told so
+  const isLoaded = useRef(false); 
+  
+  // Run this after the initial loading
   useEffect(() => {
```
Comment (levhitaIncode) [inline src/App.js:91]: borrowed the pattern mostly from the extended example

File: src/App.js [hunk 8]
```diff
@@ -66,51 +81,44 @@ function saveDeviceData() {
...
-        setScriptLoaded(false);
-      };
-
-      document.body.appendChild(script);
-
-      return () => {
-        document.body.removeChild(script);
-      };
-    }
+    // Only fetch the data if we havent fetched it yet
```
Comment (levhitaIncode) [inline src/App.js:92]: borrowed the pattern mostly from the extended example

File: .env.example [hunk 9]
```diff
@@ -1,14 +1,5 @@
 # Incode API URL for demo or saas environment 
 REACT_APP_INCODE_API_URL="https://demo-api.incodesmile.com/0"
 
-# Incode API Key. Provided by Incode
-REACT_APP_INCODE_API_KEY=""
```
Comment (levhitaIncode) [inline .env.example:5]: removed apikey requirement

File: src/App.js [hunk 10]
```diff
@@ -66,51 +81,44 @@ function saveDeviceData() {
...
+    }).catch(
+      (e)=>console.log(e)
+    );
+    
+    // We already sent the async call, don't call it again
+    isLoaded.current = true;
   }, []);
-
+  
+  useEffect(()=>{
```
Comment (levhitaIncode) [inline src/App.js:106]: this gets called after setSession is resolved at the end of the async fetch above

File: src/App.js [hunk 11]
```diff
@@ -66,51 +81,44 @@ function saveDeviceData() {
...
-      {scriptLoaded ? (
-        <div ref={incodeContainerRef}></div>
-      ) : (
-        <p>Loading the Incode Web SDK...</p>
-      )}
+    <div ref={incodeContainerRef}></div>
+    {!session && (
+      <p>Starting session...</p>
+    )}
+    <div ref={incodeContainerRef}></div>
```
Comment (levhitaIncode) [inline src/App.js:121]: I'm dumb i couldn't find a better way to keep the ref alive :V

File: src/incode.js
Comment (levhitaIncode) [inline src/incode.js:1]: identical file that other implementations

File: src/translations.js [hunk 12]
```diff
@@ -1372,3 +1372,5 @@ export default {
   },
   youllNeed: "Vas a necesitar:",
 };
+
+export default translations;
```
Comment (levhitaIncode) [inline src/translations.js:1376]: just solving a warning from the compiler

**Review by incodecota — APPROVED** [review]

**Pull request merged**